app/javascript/components/align-and-fine-tune/AlignAndFineTune.jsx

```diff
@@ -479,26 +479,32 @@ const AlignAndFineTune = (props) => {
    */
   const saveAllAlignments = async () => {
     /// Check for synthetic elements and save it if any
+    let synthetics = mappingTerms.filter(
+      (mTerm) => mTerm.type === syntheticLabel
+    );
+    let alignments = mappingTerms.filter(
+      (mTerm) => mTerm.type !== syntheticLabel
+    );
     let savedTerms = 0;
+
     await Promise.all(
-      mappingTerms
-        .filter((mTerm) => mTerm.type === syntheticLabel)
-        .map((mTerm) => {
-          saveSynthetic(mTerm);
+      synthetics.map(async (mTerm) => {
+        let added = await saveSynthetic(mTerm);
+        if (added) {
           savedTerms++;
-        })
+        }
+      })
     );
 
-    if (!errors.length) {
+    /// If we had synthetics to save and we successfully saved it
+    if (!synthetics.length || (synthetics.length && savedTerms)) {
       /// Save the rest of the changes (new alignments)
       await Promise.all(
-        mappingTerms
-          .filter((mTerm) => mTerm.type !== syntheticLabel)
-          .map((mTerm) => {
-            if (saveOneAlignment(mTerm)) {
-              savedTerms++;
-            }
-          })
+        alignments.map((mTerm) => {
+          if (saveOneAlignment(mTerm)) {
```

Comment:
`saveOneAlignment` is an async function, without `await` it'll always return a truthy value here.